scripts/validate-artifacts.ps1

```diff
@@ -34,138 +34,155 @@ Write-Output "Validating artifacts"
-    }  else { 
-        Write-Output `u{1F914}
-        throw "More than one virtual network resource was found in the task resource group. Please make sure that your script deploys only 1 virtual network, and try again."
-    }
-} else {
+if ($virtualMachine.location -eq "uksouth" ) { 
+    Write-Output "`u{2705} Checked Virtual Machine location - OK."
+} else { 
     Write-Output `u{1F914}
-    throw "Unable to find virtual network in the task resource group. Please make sure that your script creates a virtual network and try again."
+    throw "Virtual is not deployed to the UK South region. Please re-deploy VM to the UK South region and try again."
+}
+
+if (-not $virtualMachine.zones) { 
+    Write-Output "`u{2705} Checked Virtual Machine availability zone - OK."
 }
 
-$virtualNetworkName = $virtualNetwork.name.Replace("[parameters('virtualNetworks_", "").Replace("_name')]", "")
-if ($virtualNetworkName -eq "vnet") { 
-    Write-Output "`u{2705} Checked the virtual network name - OK."
+if (-not $virtualMachine.properties.securityProfile) { 
+    Write-Output "`u{2705} Checked Virtual Machine security type settings - OK."
 } else { 
     Write-Output `u{1F914}
-    throw "Unable to verify the virtual network name. Please make sure that your script creates a virtual network called 'vnet' and try again."
+    throw "Virtual machine security type is set to TMP or Confidential. Please re-deploy VM with security type set to 'Standard' and try again."
 }
 
-$subnet = $virtualNetwork.properties.subnets
-if ($subnet) {
-    if ($subnet.name.Count -eq 1) { 
-        Write-Output "`u{2705} Checked if subnet exists - OK."
-    }  else { 
-        Write-Output `u{1F914}
-        throw "More than one subnet was found in the virtual network. Please make sure that your script deploys only 1 subnet, and try again."
-    }
-} else {
+if ($virtualMachine.properties.storageProfile.imageReference.publisher -eq "canonical") { 
+    Write-Output "`u{2705} Checked Virtual Machine OS image publisher - OK" 
+} else { 
     Write-Output `u{1F914}
-    throw "Unable to find subnet in the virtual network. Please make sure that your script creates a subnet and try again."
+    throw "Virtual Machine uses OS image from unknown published. Please re-deploy the VM using OS image from publisher 'Cannonical' and try again."
 }
-
-if ($subnet.name -eq "default") { 
-    Write-Output "`u{2705} Checked the subnet name - OK."
+if ($virtualMachine.properties.storageProfile.imageReference.offer.Contains('ubuntu-server') -and $virtualMachine.properties.storageProfile.imageReference.sku.Contains('22_04')) { 
+    Write-Output "`u{2705} Checked Virtual Machine OS image offer - OK"
 } else { 
     Write-Output `u{1F914}
-    throw "Unable to verify the subnet name. Please make sure that your script creates a subnet called 'default' and try again."
+    throw "Virtual Machine uses wrong OS image. Please re-deploy VM using Ubuntu Server 22.04 and try again" 
```

Comment:
There is a typo in the error message: 'authentification' should be 'authentication'. Correcting this will make the error message clearer.